src/components/CenturyFilter/CenturyFilter.tsx

```diff
@@ -0,0 +1,70 @@
+import cn from 'classnames';
+import { FC } from 'react';
+import { Link, useSearchParams } from 'react-router-dom';
+
+export const CenturyFilter: FC = () => {
+  const [searchParams] = useSearchParams();
+
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const getSearchParamsStr = (century: string) => {
+    searchParams.delete('centuries');
+
+    if (century === '') {
+      return searchParams.toString();
+    }
+
+    const centuriesToSet = centuries.includes(century)
+      ? centuries.filter(centennial => centennial !== century)
+      : [...centuries, century];
+
+    centuriesToSet.forEach(centennial => {
+      searchParams.append('centuries', centennial);
+    });
+
+    return searchParams.toString();
+  };
+
+  const centuriesArrForRemder = ['16', '17', '18', '19', '20'];
```

Comment:
```suggestion
  const centuriesArrForRender = ['16', '17', '18', '19', '20'];
```
move it outside the component or wrap with useMemo